src/main/java/uk/nickbdyer/tictactoe/players/PlayerFactory.java

```diff
@@ -20,23 +17,15 @@ public PlayerFactory(UserInterface userInterface) {
         this.ui = userInterface;
     }
 
-    public List<Player> create(GameType type) {
-        if (type == GameType.HvsH) {
-            return Arrays.asList(new Human(X, ui), new Human(O, ui));
-        } else if (type == GameType.HvsAi) {
-            return Arrays.asList(new Human(X, ui), slow(new DumbComputer(O)));
-        } else if (type == GameType.AivsH) {
-            return Arrays.asList(slow(new DumbComputer(X)), new Human(O, ui));
-        } else if (type == GameType.AivsAi) {
-            return Arrays.asList(slow(new DumbComputer(X)), slow(new DumbComputer(O)));
-        } else if (type == GameType.HvsPAi) {
-            return Arrays.asList(new Human(X, ui), slow(new PerfectComputer(O)));
-        } else if (type == GameType.PAivsH) {
-            return Arrays.asList(slow(new PerfectComputer(X)), new Human(O, ui));
-        } else if (type == GameType.PAivsPAi) {
-            return Arrays.asList(slow(new PerfectComputer(X)), slow(new PerfectComputer(O)));
+    public Player create(PlayerType type, Mark mark) {
+        if (type == H) {
+            return new Human(mark, ui);
+        } else if (type == Ai) {
+            return new DumbComputer(mark);
+        } else if (type == PAi) {
+            return new PerfectComputer(mark);
         } else {
-            throw new InvalidGameTypeException();
+            throw new InvalidPlayerTypeException();
```

Comment:
Is this for when `type` is `TEST`?
If `PlayerType` didn't include `TEST`, the compiler would check this and would fail at compile time, not runtime.